aw_watcher_window/windows.py

```diff
@@ -0,0 +1,42 @@
+from typing import Optional
+
+import wmi
+import win32gui
+import win32process
+
+c = wmi.WMI()
+
+"""
+Much of this derived from: http://stackoverflow.com/a/14973422/965332
+"""
+
+def get_app_path(hwnd) -> Optional[str]:
+    """Get application path given hwnd."""
+    path = None
+    _, pid = win32process.GetWindowThreadProcessId(hwnd)
+    for p in c.query('SELECT ExecutablePath FROM Win32_Process WHERE ProcessId = %s' % str(pid)):
+        path = p.ExecutablePath
+        break
+    return path
+
+def get_app_name(hwnd) -> Optional[str]:
```

Comment:
Not `get_app_name`, but yeah, `get_app_path` is. But might be useful in the future, so keeping it there.